projects/admin/src/app/service/record-permission.service.ts

```diff
@@ -0,0 +1,197 @@
+        other: this._translateService.instant('has # acquisition order lines attached')
+      },
+      acq_orders: {
+        '=1': this._translateService.instant('has 1 acquisition orders attached'),
+        other: this._translateService.instant('has # acquisition orders attached')
+      },
+      acq_accounts: {
+        '=1': this._translateService.instant('has 1 acquisition account attached'),
+        other: this._translateService.instant('has # acquisition accounts attached')
+      },
+      budgets: {
+        '=1': this._translateService.instant('has 1 budget attached'),
+        other: this._translateService.instant('has # budgets attached')
+      },
+      circ_policies: {
+        '=1': this._translateService.instant('has 1 circulation policy attached'),
+        other: this._translateService.instant('has # circulation policies attached')
+      },
+      documents: {
+        '=1': this._translateService.instant('has 1 document attached'),
+        other: this._translateService.instant('has # documents attached')
+      },
+      item_types: {
+        '=1': this._translateService.instant('has 1 item type attached'),
+        other: this._translateService.instant('has # item types attached')
+      },
+      items: {
+        '=1': this._translateService.instant('has 1 item attached'),
+        other: this._translateService.instant('has # items attached')
+      },
+      libraries: {
+        '=1': this._translateService.instant('has 1 library attached'),
+        other: this._translateService.instant('has # libraries attached')
+      },
+      loans: {
+        '=1': this._translateService.instant('has 1 loan attached'),
+        other: this._translateService.instant('has # loans attached')
+      },
+      locations: {
+        '=1': this._translateService.instant('has 1 location attached'),
+        other: this._translateService.instant('has # locations attached')
+      },
+      organisations: {
+        '=1': this._translateService.instant('has 1 organisation attached'),
+        other: this._translateService.instant('has # organisations attached')
+      },
+      patron_types: {
+        '=1': this._translateService.instant('has 1 patron type attached'),
+        other: this._translateService.instant('has # patron types attached')
+      },
+      patrons: {
+        '=1': this._translateService.instant('has 1 patron attached'),
+        other: this._translateService.instant('has # patrons attached')
+      }
+    };
+  }
+
+  /**
+   * Others messages
+   * @return array
+   */
+  private othersMessages() {
+    return {
+      is_default: this._translateService.instant('The default record cannot be deleted'),
+      has_settings: this._translateService.instant('The record contains settings'),
+      harvested: this._translateService.instant('The record has been harvested')
+    };
+  }
+}
+
+/**
+ * Permission response structure
+ */
+export interface RecordPermissionInterface {
```

Comment:
As I know an Interface name does not contains the Interface string in angular. For example: https://angular.io/api/forms/ControlValueAccessor or https://angular.io/tutorial/toh-pt1 (interface Hero).

Reply:
done